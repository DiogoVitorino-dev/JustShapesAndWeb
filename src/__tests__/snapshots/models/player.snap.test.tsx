import { render, renderHook } from "@testing-library/react-native";
import { useSharedValue } from "react-native-reanimated";

import Player, { PlayerData } from "@/models/player";

describe("Player model - Snapshot test", () => {
  it("Should renders correctly", () => {
    const data = renderHook(() => useSharedValue<PlayerData>({ x: 25, y: 50 }));

    const tree = render(<Player data={data.result.current} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
