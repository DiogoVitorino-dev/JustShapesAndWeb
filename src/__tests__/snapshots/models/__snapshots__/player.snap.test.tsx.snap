// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player model - Snapshot test Should renders correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "height": 20,
        "left": 25,
        "top": 50,
        "transform": [
          {
            "rotate": "0deg",
          },
          {
            "scaleY": 1,
          },
          {
            "scale": 1,
          },
        ],
        "width": 20,
      },
    }
  }
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(20, 255, 170)",
      "borderRadius": 2,
      "height": 20,
      "left": 25,
      "position": "absolute",
      "top": 50,
      "transform": [
        {
          "rotate": "0deg",
        },
        {
          "scaleY": 1,
        },
        {
          "scale": 1,
        },
      ],
      "width": 20,
    }
  }
  testID="playerModel"
>
  <View
    animatedStyle={
      {
        "value": {
          "position": "absolute",
        },
      }
    }
    collapsable={false}
    style={
      {
        "position": "absolute",
      }
    }
  >
    <View
      animatedStyle={
        {
          "value": {
            "backgroundColor": "rgb(20, 255, 170)",
            "height": 4.222222222222222,
            "opacity": 0,
            "right": 0,
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "width": 4.222222222222222,
          },
        }
      }
      collapsable={false}
      style={
        {
          "backgroundColor": "rgb(20, 255, 170)",
          "height": 4.222222222222222,
          "opacity": 0,
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateX": 0,
            },
          ],
          "width": 4.222222222222222,
        }
      }
      testID="PlayerMovementParticle"
    />
    <View
      animatedStyle={
        {
          "value": {
            "backgroundColor": "rgb(20, 255, 170)",
            "height": 2.2222222222222223,
            "opacity": 0,
            "right": 0,
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "width": 2.2222222222222223,
          },
        }
      }
      collapsable={false}
      style={
        {
          "backgroundColor": "rgb(20, 255, 170)",
          "height": 2.2222222222222223,
          "opacity": 0,
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateX": 0,
            },
          ],
          "width": 2.2222222222222223,
        }
      }
      testID="PlayerMovementParticle"
    />
    <View
      animatedStyle={
        {
          "value": {
            "backgroundColor": "rgb(20, 255, 170)",
            "height": 5.222222222222222,
            "opacity": 0,
            "right": 0,
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "width": 5.222222222222222,
          },
        }
      }
      collapsable={false}
      style={
        {
          "backgroundColor": "rgb(20, 255, 170)",
          "height": 5.222222222222222,
          "opacity": 0,
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateX": 0,
            },
          ],
          "width": 5.222222222222222,
        }
      }
      testID="PlayerMovementParticle"
    />
  </View>
</View>
`;
